smqtk_detection/impls/detect_image_objects/resnet_frcnn.py

```diff
@@ -0,0 +1,171 @@
+import numpy as np
+
+import torch
+import torchvision.models as models
+from torchvision import transforms
+from torchvision.ops import boxes as box_ops
+import torch.nn.functional as F
+from torchvision.models.detection.roi_heads import RoIHeads
+
+from typing import Tuple, Iterable, Dict, Hashable, List
+
+from smqtk_detection.interfaces.detect_image_objects import DetectImageObjects
+from smqtk_detection.utils.bbox import AxisAlignedBoundingBox
+
+import warnings
+warnings.filterwarnings("ignore", category=UserWarning)
+
+
+class ResNetFRCNN(DetectImageObjects):
+    """
+    ``DetectImageObjects`` implementation using ``torchvision``'s Faster R-CNN
+    with a ResNet-50-FPN backbone, pretrained on COCO train2017.
+    """
+
+    def __init__(
+        self,
+        box_thresh: float = 0.05,
+        num_dets: int = 100,
+        img_batch_size: int = 1,
+        use_cuda: bool = True,
+    ):
+
+        self.box_thresh = box_thresh
+        self.num_dets = num_dets
+        self.img_batch_size = img_batch_size
+
+        self.model = models.detection.fasterrcnn_resnet50_fpn(
+            pretrained=True,
+            progress=False,
+            box_detections_per_img=num_dets,
+            box_score_thresh=box_thresh
+        )
+
+        self.model = self.model.eval()
+
+        if torch.cuda.is_available() and use_cuda:
+            self.use_cuda = True
+            self.model = self.model.cuda()
+        else:
+            self.use_cuda = False
+
+        self.model_loader = transforms.Compose([
+            transforms.ToPILImage(),
+            transforms.ToTensor(),
+        ])
+
+        RoIHeads.postprocess_detections = postprocess_detections
+
+    @torch.no_grad()
+    def detect_objects(
+        self,
+        img_iter: Iterable[np.ndarray]
+    ) -> Iterable[Iterable[Tuple[AxisAlignedBoundingBox, Dict[Hashable, float]]]]:
+
+        formatted_dets = []       # AxisAlignedBoundingBox detections to return
+
+        img_tensors = [self.model_loader(img) for img in img_iter]
+
+        if self.use_cuda:
+            img_tensors = [tensor.cuda() for tensor in img_tensors]
+
+        # split into batches
+        batches = []
+        for i in range(0, len(img_tensors), self.img_batch_size):
+            batches.append(img_tensors[i : i + self.img_batch_size])
+
+        all_img_dets = sum([self.model(batch) for batch in batches], [])  # type: List[Dict]
+
+        for img_dets in all_img_dets:
+
+            bboxes = img_dets['boxes'].cpu().numpy()
+            scores = img_dets['scores'].cpu().numpy()
+
+            a_bboxes = [AxisAlignedBoundingBox(
+                [box[0], box[1]], [box[2], box[3]]
+            ) for box in bboxes]
+
+            score_dicts = []
+
+            for img_scores in scores:
+                score_dict = {}
```

Comment:
I have locally confirmed that your typechecking error for L98 is due to not annotating this dict. Without an explicit annotation, mypy is left to implicitly assign one based on its immediate usage, which is where the `Dict[int, Any]` comes from. Annotating this as `Dict[Hashable, float]` is compatible with the assignments happening here as well as resolves the error being raised about L98.